spec/requests/new_document_spec.rb

```diff
@@ -1,64 +1,58 @@
 
-    it "redirects when a supertype managed elsewhere is selected" do
-      supertype = Supertype.all.find(&:managed_elsewhere)
-      get choose_document_type_path, params: { supertype: supertype.id }
-      expect(response).to redirect_to(supertype.managed_elsewhere_url)
-    end
-
-    it "redirects to guidance page when \"I'm not sure\" option is selected" do
-      supertype = Supertype.all.find { |type| type.id == "not-sure" }
-      get choose_document_type_path, params: { supertype: supertype.id }
-      expect(response).to redirect_to(guidance_url)
+    it "shows the page for the selected document type" do
+      get new_document_path, params: { type: "news" }
+      expect(response).to have_http_status(:ok)
+      expect(response.body).to have_content("News story")
     end
 
-    it "returns an unprocessable response with an issue when a supertype isn't selected" do
-      get choose_document_type_path
-      expect(response).to have_http_status(:unprocessable_entity)
-      expect(response.body)
-        .to have_content(I18n.t!("requirements.supertype.not_selected.form_message"))
+    it "returns a 404 when the requested document type selection doesn't exist" do
+      get new_document_path, params: { type: "foo" }
+      expect(response.status).to eq(404)
     end
   end
 
-  describe "POST /documents/create" do
-    let(:supertype_id) { Supertype.all.first.id }
-
+  describe "POST /documents/select" do
     it "redirects to document edit content when a content publisher managed document type is selected" do
-      document_type = build(:document_type)
-      post create_document_path,
-           params: { supertype: supertype_id, document_type: document_type.id }
+      post new_document_path, params: { type: "news", selected_option_id: "news_story" }
 
       expect(response).to redirect_to(content_path(Document.last))
       follow_redirect!
-      expect(response.body).to have_content(document_type.label.downcase)
+      expect(response.body).to have_content("news story")
     end
 
     it "redirects when a document type managed elsewhere is selected" do
-      document_type = build(
-        :document_type,
-        managed_elsewhere: { "hostname" => "whitehall",
-                             "path" => "/document-type" },
-      )
-      post create_document_path,
-           params: { supertype: supertype_id, document_type: document_type.id }
+      post new_document_path, params: { type: "root", selected_option_id: "not_sure" }
 
-      expect(response)
-        .to redirect_to("https://whitehall.test.gov.uk/document-type")
+      expect(response).to redirect_to(guidance_url)
+    end
+
+    it "asks the user to refine their selection when the document type has subtypes" do
+      post new_document_path, params: { type: "root", selected_option_id: "news" }
+
+      expect(response).to redirect_to(new_document_path(type: "news"))
+      follow_redirect!
+      expect(response.body).to have_content("News story")
     end
```

Comment:
I think rather than having 3 tests here for the different types of redirect we only need to have one since the controller we're testing only has logic for one type of redirect_url and we're testing how those are set in the interactor.  I'm guessing though that this came about when we had multiple conditions in the controller when it definitely made sense to be here. 

But yeah I think now it probably makes most sense for it to be "it redirects users to a destination on a successful request"

Edit: If you go with the suggestion I've made about dealing with redirect destination in controller we'd want to keep the 3 test approach.

Reply:
I'm going with the suggestion about dealing with the redirect in the controller. It was my preferred option as it felt wrong to have to include `url_helpers` in the interactor.